modules/dispatcher/dispatcher.c

```diff
@@ -1360,15 +1362,34 @@ mi_response_t *ds_mi_reload_1(const mi_params_t *params,
 {
 	ds_partition_t *partition;
 	str partname;
+	str inherit_state;
+	int is_inherit_state = INHERIT_STATE_YES;
 
-	if (get_mi_string_param(params, "partition", &partname.s, &partname.len) < 0)
-		return init_mi_param_error();
+	if (get_mi_string_param(params, "partition", &partname.s, &partname.len) < 0) {
```

Comment:
Not sure if this was in the original PR, but it is not a good idea to fall back to "default" on error here, if may have unpredictable side effects, please take this change out